test/integration/channels.js

```diff
@@ -2659,6 +2702,66 @@ describe('Quote messages', () => {
 			expect(clm.quoted_message.user).to.not.be.undefined;
 		});
 
+		it('is possible to otherwise change a message that contains a quoted_message', async () => {
+			const msgText =
+				'The first message that quotes a message, now with changed text';
+
+			let messageUpdatedListenerRes;
+			const messageUpdatedListener = new Promise((res) => {
+				messageUpdatedListenerRes = res;
+			});
+
+			const webhook = await setupWebhook(
+				client,
+				'webhook_url',
+				(request, body, response) => {
+					const b = JSON.parse(body);
+					if (b.type === 'message.updated') {
+						messageUpdatedListenerRes(b);
+					}
+					response.writeHead(200);
+				},
+			);
+
+			const updateMessageReq = client.updateMessage(
+				{
+					id: messageWithQuote.id,
+					text: msgText,
+				},
+				friend,
+			);
+
+			const [updatedEvent, res] = await Promise.all([
+				messageUpdatedListener,
+				updateMessageReq,
+			]);
+
+			expect(res.message).to.not.be.undefined;
+			expect(res.message.id).to.not.be.empty;
+			expect(res.message.text).to.equal(msgText);
+			expect(res.message.quoted_message).to.not.be.undefined;
+			expect(res.message.quoted_message_id).to.equal(secondMessage.id);
+			expect(res.message.quoted_message.id).to.equal(secondMessage.id);
+			expect(res.message.quoted_message.text).to.equal(secondMessage.text);
+			expect(res.message.type).to.equal('regular');
+			expect(res.message.parent_id).to.be.undefined;
+			expect(res.message.quoted_message.user).to.not.be.undefined;
+
+			expect(updatedEvent.message).to.not.be.undefined;
+			expect(updatedEvent.message.id).to.not.be.empty;
+			expect(updatedEvent.message.text).to.equal(msgText);
+			expect(updatedEvent.message.quoted_message).to.not.be.undefined;
+			expect(updatedEvent.message.quoted_message_id).to.equal(secondMessage.id);
+			expect(updatedEvent.message.quoted_message.id).to.equal(secondMessage.id);
+			expect(updatedEvent.message.quoted_message.text).to.equal(secondMessage.text);
+			expect(updatedEvent.message.type).to.equal('regular');
+			expect(updatedEvent.message.parent_id).to.be.undefined;
+			expect(updatedEvent.message.quoted_message.user).to.not.be.undefined;
+
+			console.log('before teardown');
```

Comment:
is this necessary or was it just for local testing purposes?

Reply:
Good catch, will remove